### PR TITLE
Add support for ConnectionTimeout and OperationTimeout config & Use logger from context

### PR DESCRIPTION
**Pull request by milanbhagwat**

### CODE REVIEW

File: pulsar/connection/connection.go [hunk 1]
```diff
@@ -32,6 +33,8 @@ type Settings struct {
 	AthenzAuthentication map[string]string `md:"athenzAuth"`
 	JWT                  string            `md:"jwt"`
 	AllowInsecure        bool              `md:"allowInsecure"`
+	ConnectionTimeout    int               `md:connTimeout`
```
Comment (vijaynalawade) [inline pulsar/connection/connection.go]: I think connection and operation timeout type can be **int8**.

**Review by vijaynalawade — CHANGES_REQUESTED** [review]

**Review by vijaynalawade — APPROVED** [review]

File: pulsar/activity/publish/activity.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ import (
 	"github.com/project-flogo/core/support/log"
 )
 
-var logger = log.ChildLogger(log.RootLogger(), "pulsar-publish")
+var logger log.Logger
```
Comment (lixingwang) [inline pulsar/activity/publish/activity.go]: Remove logger from global,  we don't need this to be global right?

**Review by lixingwang — APPROVED** [review]
Please just remove logger from global variable other part looks good to me

File: pulsar/activity/publish/activity.go (hunk 2 above)
Comment (vijaynalawade) [reply pulsar/activity/publish/activity.go]: Agree. This is not needed.

Comment (milanbhagwat) [reply pulsar/activity/publish/activity.go]: Changes Done

**Review by vijaynalawade — APPROVED** [review]

**Review by lixingwang — APPROVED** [review]

**Pull request merged**